src/Components/ContextMenu.re

```diff
@@ -0,0 +1,257 @@
+open Oni_Core;
+
+open Revery;
+open Revery.UI;
+open Revery.UI.Components;
+
+module Option = Utility.Option;
+
+module Constants = {
+  let menuWidth = 200;
+  // let maxMenuHeight = 600;
+};
+
+// TYPES
+
+module Id: {
+  type t;
+  let create: unit => t;
+} = {
+  type t = int;
+
+  let lastId = ref(0);
+  let create = () => {
+    incr(lastId);
+    lastId^;
+  };
+};
+
+[@deriving show({with_path: false})]
+type item('data) = {
+  label: string,
+  // icon: option(IconTheme.IconDefinition.t),
```

Comment:
`IconTheme` currently lives in `Model`. I'm not sure if `Core` is the right place for it, so for now we'll have to do without.